### ISSUE TITLE
ProxyShape plays only once for animated USD mesh

### ISSUE THREAD

**Issue by BigRoy** [Autodesk/maya-usd#126]
### Issue

When playing through time with an animated geometry the first playback through time seems smooth with the attached USD pighead. It gives *no visible errors* as to why it stops playing back. Additionally I've had it happen that when selecting the mesh the wireframe would actually play correctly, but then after the first cycle there also stop playing altogether. And then, when going to another frame and reselecting the mesh shows the updated "wireframe selection overlay". This seems to be a display update bug in Viewport 2.0.

[pighead_mesh.zip](https://github.com/Autodesk/maya-usd/files/3875737/pighead_mesh.zip)

_The attached USD scene is an export from Houdini._

It works fine however when playing back other animated production `.abc` files I had laying around, with those I can *not* reproduce it.

### To reproduce

This is easily visible when loaded through `AL_USDMayaPlugin` with the menu item but the same occurs through the `mayaUsdProxyShape` with this:
```python
from maya import cmds

path = "path/to/pighead_mesh.usd"
node = cmds.createNode("mayaUsdProxyShape")
cmds.setAttr(node + ".filePath", path, type="string")

# Make sure to connect time
cmds.connectAttr("time1.outTime", node + ".time")
```

It seems to fail as soon as you "go back in frames" and especially whenever you've played this file until the end of the frame samples, e.g. play all frames past frame `50` and then restart. Then you'll see no motion whatsoever in the Viewport.

**What I've tried:**

- Running `ogs -reset` does resolve it, until it reaches the end again.
- I have Maya 2019 timeline caching disabled. Enabling it makes no difference in behavior.
- It happens in both parallel and DG evaluation mode.
- Like the USD file my Maya scene is 24 FPS, that matches fine. With other FPS settings in Maya the issue is also there.
- When enabling debugging features from the AL USD menu there's little new information I can see that might cause it. With `USD_VALUE_RESOLUTION` it does show it consistently trying to resolve the time samples as I step through time. Like:
```
RESOLVE: reading field /usdexport4/mesh_0.points:timeSamples from @p:/Projects/test1/assets/usd_assets/work/model/houdini/OUTPUT/usd/pighead/pighead_model_full.usd@, with requested time = 18.000 (local time = 18.000) reading from sample 18.000 
RESOLVE: reading field /usdexport4/mesh_0.primvars:st:timeSamples from @p:/Projects/test1/assets/usd_assets/work/model/houdini/OUTPUT/usd/pighead/pighead_model_full.usd@, with requested time = 18.000 (local time = 18.000) reading from sample 18.000 
RESOLVE: reading field /usdexport4/mesh_0.extent:timeSamples from @p:/Projects/test1/assets/usd_assets/work/model/houdini/OUTPUT/usd/pighead/pighead_model_full.usd@, with requested time = 18.000 (local time = 18.000) reading from sample 18.000 
```
- Similarly `USDIMAGING_UPDATE` seems to trigger something:
```
[Sync] PREP: </usdexport4/mesh_0> dirtyFlags: 0x80006a [<Varying> Extent Points Primvar CustomBits:10000000]
[Update] Update for time (37.000000)
```
- And `USDIMAGING_CHANGES`:
```
[UpdateForTime] Mesh path: </usdexport4/mesh_0>
```

### Specs

- `dev` branch with [commit `aa41c55`](https://github.com/Autodesk/maya-usd/commit/aa41c5554e4a3c420989350b2503fc3ca6027421)
- Maya 2019.2
- Windows 10
- Pixar USD dev ( `b85ddac2` )

**Comment by kxl-adsk** [comment]
Can you update Pixar USD to commit mentioned in https://github.com/Autodesk/maya-usd/blob/dev/doc/DEVELOPER.md ?
Also, this is using VP2RenderDelegate or without?

**Comment by HdC-adsk** [comment]
It is VP2RenderDelegate only. But I am not sure if it is a regression or not. Did it work before? @BigRoy

**Comment by HdC-adsk** [comment]
I've confirmed it is a regression. We tried to avoid the expensive setGeometryForRenderItem() call, but the side effect of that function only include re-consolidation. I will need to find a cheaper way to trigger reconsolidation when the geometry is changed.

**Comment by BigRoy** [comment]
> I've confirmed it is a regression. 

Thanks for confirming. I'm assuming you won't need more information from my end as you have been able to reproduce it.

It does still feel odd that the "bug" does not appear when I load animated `.abc` files into the Proxy shape. What could be the reason behind that? Does it load it in a very different manner?

**Comment by HdC-adsk** [comment]
Thanks. There can be chance that deformed shapes also have other attributes being dirtied which can trigger reconsolidation, thus I didn't find this regression when testing the optimization change without the deformation scene I have. But I don't think we can load abc file into the proxy shape, how do you do it?

**Comment by BigRoy** [comment]
> But I don't think we can load abc file into the proxy shape, how do you do it?

It seems you can, works fine in `mayaUsdProxyShape`. Just like in `usdview`. :) Not sure how, but I considered it a nice feature as I was testing some stuff. Definitely give it a go.

**Comment by HdC-adsk** [comment]
Logged as MAYA-102223.

**Comment by HdC-adsk** [comment]
@BigRoy FYI, I've created PR #139 for this which is currently under review. It would be perfect if you can run a test with that branch for double confirm. Thanks.

**Comment by BigRoy** [comment]
Thanks. I tried to give your branch a go but unfortunately it fails to compile:

```
    Building with settings:
      Source directory          D:\dev\usd\maya-usd
      Workspace directory       D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\tmp
      Build directory           D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd
      Install directory         C:\Users\Roy\packages\maya_usd\0.0.git\platform-windows\arch==AMD64
      Variant                   RelWithDebInfo
      CMake generator           Default
      Build Log                 D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\build_log.txt
      Stages arguments          ['clean', 'configure', 'build', 'install']
2019-11-29 16:20
cmake -DCMAKE_INSTALL_PREFIX="C:\Users\Roy\packages\maya_usd\0.0.git\platform-windows\arch==AMD64" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -G "Visual Studio 15 2017 Win64" -DMAYA_LOCATION="C:\Program Files\Autodesk\Maya2019" -DPXR_USD_LOCATION="D:\rez_packages\usd\19.11\platform-windows\PySide" -DMAYA_DEVKIT_LOCATION="C:\Users\Roy\packages\maya_devkit\2019.2\platform-windows\devkitBase" "D:\dev\usd\maya-usd"
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.18362.
-- The C compiler identification is MSVC 19.16.27034.0
-- The CXX compiler identification is MSVC 19.16.27034.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: C:/Users/Roy/packages/python/2.7.16/platform-windows/arch-AMD64/app/python.exe (found suitable version "2.7.16", minimum required is "2.7")
-- _PYTHON_VERSION_LIST: 2.7.16.final.0
-- PYTHON_PREFIX: C:\Users\Roy\packages\python\2.7.16\platform-windows\arch-AMD64\app
-- PYTHON_INCLUDE_DIR: C:\Users\Roy\packages\python\2.7.16\platform-windows\arch-AMD64\app\include
-- PYTHON_SITE_PACKAGES: C:\Users\Roy\packages\python\2.7.16\platform-windows\arch-AMD64\app\Lib\site-packages
-- PYTHON_MODULE_EXTENSION: .pyd
-- PYTHON_IS_DEBUG: 0
-- PYTHON_SIZEOF_VOID_P: 8
-- PYTHON_LIBRARY_SUFFIX: 27
-- PYTHON_LIBDIR:
-- PYTHON_MULTIARCH:
-- Found PythonLibs: C:/Users/Roy/packages/python/2.7.16/platform-windows/arch-AMD64/app/libs/Python27.lib
-- Found Maya: C:/Program Files/Autodesk/Maya2019/bin/maya.exe (found version "20190200")
-- USD include dir: D:/rez_packages/usd/19.11/platform-windows/PySide/include
-- USD library dir: D:/rez_packages/usd/19.11/platform-windows/PySide/lib
-- USD version: 0.19.11
-- Found USD: D:\rez_packages\usd\19.11\platform-windows\PySide (found version "0.19.11")
Updated PYTHONPATH: C:\Users\Roy\packages\PyOpenGL\3.1.0\python-2/python;C:\Users\Roy\packages\Jinja2\2.10.3/python;C:\Users\Roy\packages\MarkupSafe\1.1.1\os-windows-10\python-2.7/python;C:\Users\Roy\packages\PySide\1.2.4\os-windows-10\python-2.7/python;;D:\rez_packages\usd\19.11\platform-windows\PySide/lib/python;D:\rez_packages\usd\19.11\platform-windows\PySide\lib\python
Updated PATH: C:\Users\Roy\packages\nasm\2.14.02\platform-windows\arch-AMD64/app/rdoff;C:\Users\Roy\packages\nasm\2.14.02\platform-windows\arch-AMD64/app/;C:\Users\Roy\packages\PyOpenGL\3.1.0\python-2/bin;C:\Users\Roy\packages\Jinja2\2.10.3/bin;C:\Users\Roy\packages\MarkupSafe\1.1.1\os-windows-10\python-2.7/bin;C:\Users\Roy\packages\PySide\1.2.4\os-windows-10\python-2.7/bin;C:\Users\Roy\packages\cmake\3.14.4\platform-windows\arch-AMD64/app/bin;C:\Users\Roy\packages\python\2.7.16\platform-windows\arch-AMD64/app/;C:\Users\Roy\packages\python\2.7.16\platform-windows\arch-AMD64/app/Lib\idlelib;C:\Windows;C:\Windows\system32;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\bin\Roslyn;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\;C:\Program Files (x86)\Microsoft SDKs\Windows\v8.1A\bin\NETFX 4.5.1 Tools\;C:\Program Files (x86)\Windows Kits\10\bin\10.0.17763.0\x86;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\\MSBuild\15.0\bin;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;D:\rez_packages\usd\19.11\platform-windows\PySide/bin;D:\rez_packages\usd\19.11\platform-windows\PySide/lib/;C:/Program Files/Common Files/Autodesk Shared/;C:/Program Files (x86)/Autodesk/Backburner/;C:/Program Files/Autodesk/Maya2019/bin;D:\rez_packages\usd\19.11\platform-windows\PySide\bin;D:\rez_packages\usd\19.11\platform-windows\PySide\lib
-- UFE include dir: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/include
-- UFE library: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/lib/ufe_1.lib
-- UFE version: 1.0.0
-- Building with UFE 1.0.0 features enabled.
-- ========== Installing GoogleTest... ==========
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.18362.
-- Configuring done
-- Generating done
-- Build files have been written to: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-config
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-config/CMakeFiles/generate.stamp is up-to-date.
  Creating directories for 'googletest'
  Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-config/CMakeLists.txt
  CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-config/CMakeFiles/generate.stamp is up-to-date.
  Performing download step (git clone) for 'googletest'
  Cloning into 'googletest-src'...
  Already on 'master'
  Your branch is up-to-date with 'origin/master'.
  Performing update step for 'googletest'
  Current branch master is up to date.
  No patch step for 'googletest'
  Performing configure step for 'googletest'
  -- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.18362.
  -- The C compiler identification is MSVC 19.16.27034.0
  -- The CXX compiler identification is MSVC 19.16.27034.0
  -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
  -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
  -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found PythonInterp: C:/Users/Roy/packages/python/2.7.16/platform-windows/arch-AMD64/app/python.exe (found version "2.7.16")
  -- Looking for pthread.h
  -- Looking for pthread.h - not found
  -- Found Threads: TRUE
  -- Configuring done
  -- Generating done
CUSTOMBUILD : CMake warning :  [D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-config\googletest.vcxproj]
    Manually-specified variables were not used by the project:

      CMAKE_BUILD_TYPE
      FORCE_SHARED_CRT


  -- Build files have been written to: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build
  Performing build step for 'googletest'
  Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

    Checking Build System
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/CMakeFiles/generate.stamp is up-to-date.
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/googlemock/CMakeFiles/generate.stamp is up-to-date.
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
    Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-src/googlemock/CMakeLists.txt
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/googlemock/CMakeFiles/generate.stamp is up-to-date.
    gtest-all.cc
    gmock-all.cc
    Generating Code...
    gmock.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gmock.lib
    Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-src/googlemock/CMakeLists.txt
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/googlemock/CMakeFiles/generate.stamp is up-to-date.
    gtest-all.cc
    gmock-all.cc
    gmock_main.cc
    Generating Code...
    gmock_main.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gmock_main.lib
    Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-src/googletest/CMakeLists.txt
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
    gtest-all.cc
    gtest.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gtest.lib
    Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-src/googletest/CMakeLists.txt
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/googletest/CMakeFiles/generate.stamp is up-to-date.
    gtest_main.cc
    gtest_main.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gtest_main.lib
    Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-src/CMakeLists.txt
    CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-build/CMakeFiles/generate.stamp is up-to-date.
  Performing install step for 'googletest'
  Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

    gmock.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gmock.lib
    gmock_main.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gmock_main.lib
    gtest.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gtest.lib
    gtest_main.vcxproj -> D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\googletest-build\lib\RelWithDebInfo\gtest_main.lib
    -- Install configuration: "RelWithDebInfo"
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-actions.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-cardinalities.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-function-mocker.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-generated-actions.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-generated-actions.h.pump
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-generated-function-mockers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-generated-function-mockers.h.pump
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-generated-matchers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-generated-matchers.h.pump
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-matchers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-more-actions.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-more-matchers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-nice-strict.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock-spec-builders.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/gmock.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/custom
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/custom/gmock-generated-actions.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/custom/gmock-generated-actions.h.pump
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/custom/gmock-matchers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/custom/gmock-port.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/custom/README.md
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/gmock-internal-utils.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/gmock-port.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gmock/internal/gmock-pp.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gmock.lib
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gmock_main.lib
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gmock.pdb
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gmock_main.pdb
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/pkgconfig/gmock.pc
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/pkgconfig/gmock_main.pc
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/cmake/GTest/GTestTargets.cmake
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/cmake/GTest/GTestTargets-relwithdebinfo.cmake
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/cmake/GTest/GTestConfigVersion.cmake
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/cmake/GTest/GTestConfig.cmake
    -- Up-to-date: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-death-test.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-matchers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-message.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-param-test.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-printers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-spi.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-test-part.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest-typed-test.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest_pred_impl.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/gtest_prod.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/custom
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/custom/gtest-port.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/custom/gtest-printers.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/custom/gtest.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/custom/README.md
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-death-test-internal.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-filepath.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-internal.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-param-util.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-port-arch.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-port.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-string.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/include/gtest/internal/gtest-type-util.h
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gtest.lib
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gtest_main.lib
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gtest.pdb
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gtest_main.pdb
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/pkgconfig/gtest.pc
    -- Installing: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/pkgconfig/gtest_main.pc
  Completed 'googletest'
  Building Custom Rule D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-config/CMakeLists.txt
  CMake does not need to re-run because D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-config/CMakeFiles/generate.stamp is up-to-date.
-- ========== ...  GoogleTest installed. ==========
CMake Warning (dev) at cmake/Googletest.cmake:53 (set):
  implicitly converting 'path' to 'STRING' type.
Call Stack (most recent call first):
  CMakeLists.txt:104 (fetch_googletest)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   filesystem
--   system
-- promoting: D:/dev/usd/maya-usd/lib/base/mayaUsd.h.src
-- promoting: D:/dev/usd/maya-usd/lib/base/api.h
-- promoting: D:/dev/usd/maya-usd/lib/base/debugCodes.h
-- promoting: D:/dev/usd/maya-usd/lib/utils/colorSpace.h
-- promoting: D:/dev/usd/maya-usd/lib/utils/query.h
-- promoting: D:/dev/usd/maya-usd/lib/utils/util.h
-- promoting: D:/dev/usd/maya-usd/lib/utils/utilFileSystem.h
-- promoting: D:/dev/usd/maya-usd/lib/utils/stageCache.h
-- promoting: D:/dev/usd/maya-usd/lib/nodes/usdPrimProvider.h
-- promoting: D:/dev/usd/maya-usd/lib/nodes/proxyShapeBase.h
-- promoting: D:/dev/usd/maya-usd/lib/nodes/proxyShapePlugin.h
-- promoting: D:/dev/usd/maya-usd/lib/nodes/stageData.h
-- promoting: D:/dev/usd/maya-usd/lib/listeners/stageNoticeListener.h
-- promoting: D:/dev/usd/maya-usd/lib/listeners/notice.h
-- promoting: D:/dev/usd/maya-usd/lib/listeners/proxyShapeNotice.h
-- promoting: D:/dev/usd/maya-usd/lib/render/vp2RenderDelegate/proxyRenderDelegate.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/Global.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/Utils.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/ProxyShapeHandler.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/ProxyShapeHierarchy.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/ProxyShapeHierarchyHandler.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/StagesSubject.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdHierarchy.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdHierarchyHandler.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdRootChildHierarchy.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdRotatePivotTranslateUndoableCommand.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdRotateUndoableCommand.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdScaleUndoableCommand.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdSceneItem.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdSceneItemOps.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdSceneItemOpsHandler.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdStageMap.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdTransform3d.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdTransform3dHandler.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdTranslateUndoableCommand.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdUndoDeleteCommand.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdUndoDuplicateCommand.h
-- promoting: D:/dev/usd/maya-usd/lib/ufe/UsdUndoRenameCommand.h
========== Pixar USD Plugin ==========
-- UFE include dir: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/include
-- UFE library: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/lib/ufe_1.lib
-- UFE version: 1.0.0
-- Building with UFE 1.0.0 features enabled.
-- Skipping validation of Maya fragment XML files because PXR_VALIDATE_GENERATED_CODE=OFF
========== AL USD Plugin ==========
CMake Deprecation Warning at plugin/al/CMakeLists.txt:7 (cmake_policy):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Found PythonInterp: C:/Users/Roy/packages/python/2.7.16/platform-windows/arch-AMD64/app/python.exe (found version "2.7.16")
-- UFE include dir: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/include
-- UFE library: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/lib/ufe_1.lib
-- UFE version: 1.0.0
-- Building with UFE 1.0.0 features enabled.
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   python
--   thread
--   filesystem
--   chrono
--   date_time
--   system
--   atomic
-- Found GTest: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gtest.lib
CMake Error at plugin/al/usdtransaction/AL/usd/transaction/tests/CMakeLists.txt:50 (set_tests_properties):
  set_tests_properties called with incorrect number of arguments.


-- Building AL_USDMayaSchemas
Processing schema classes:
ALModelAPI, ALMayaReference, ALFrameRange
Loading Templates from d:\rez_packages\usd\19.11\platform-windows\PySide\lib\usd\usd\resources\codegenTemplates
Writing API:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\api.h
Writing Schema Tokens:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\tokens.h
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\tokens.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\wrapTokens.cpp
Generating Classes:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\ModelAPI.h
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\ModelAPI.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\wrapModelAPI.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\MayaReference.h
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\MayaReference.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\wrapMayaReference.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\FrameRange.h
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\FrameRange.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\wrapFrameRange.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\plugInfo.json
Generating Schematics:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\maya\generatedSchema.usda
Processing schema classes:
ALExamplePolyCubeNode
Loading Templates from d:\rez_packages\usd\19.11\platform-windows\PySide\lib\usd\usd\resources\codegenTemplates
Writing API:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\api.h
Writing Schema Tokens:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\tokens.h
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\tokens.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\wrapTokens.cpp
Generating Classes:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\ExamplePolyCubeNode.h
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\ExamplePolyCubeNode.cpp
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\wrapExamplePolyCubeNode.cpp
        unchanged D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\plugInfo.json
Generating Schematics:
            wrote D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\plugin\al\schemas\AL\usd\schemas\mayatest\generatedSchema.usda
-- Building AL_USDMayaTranslators
========== ADSK USD Plugin ==========
-- UFE include dir: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/include
-- UFE library: C:/Users/Roy/packages/maya_devkit/2019.2/platform-windows/devkitBase/devkit/ufe/lib/ufe_1.lib
-- UFE version: 1.0.0
-- Building with UFE 1.0.0 features enabled.
-- Configuring incomplete, errors occurred!
See also "D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/CMakeFiles/CMakeOutput.log".
See also "D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/CMakeFiles/CMakeError.log".
```

Not sure why it suddenly fails to build.

[build_log.txt](https://github.com/Autodesk/maya-usd/files/3905788/build_log.txt)

Investigating now.

**Comment by HdC-adsk** [comment]
@BigRoy From the build log, I cannot find info why it failed. Did you figure out why?

**Comment by BigRoy** [comment]
The error it failed on was this I believe:
```
-- Found GTest: D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/googletest-install/lib/gtest.lib
CMake Error at plugin/al/usdtransaction/AL/usd/transaction/tests/CMakeLists.txt:50 (set_tests_properties):
  set_tests_properties called with incorrect number of arguments.
```

Why it failed I still haven't figured out.

**Comment by HdC-adsk** [comment]
Did you try a clean build?

**Comment by BigRoy** [comment]
Tried it again just now, cleared the build and install folder and tried to rebuild. No luck.

**Comment by fowlertADSK** [comment]
I just grabbed the same branch and built without error.  Currently trying to diff our outputs to see what the difference might be.

**Comment by fowlertADSK** [comment]
Can you attach those last two logs?  These ones...

See also "D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/CMakeFiles/CMakeOutput.log".
See also "D:/git_pipeline/rez-packages/maya_usd/build/platform-windows/arch==AMD64/maya-usd/CMakeFiles/CMakeError.log".

**Comment by BigRoy** [comment]
Sure can.

From `D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\CMakeFiles`:
[CMakeOutput.log](https://github.com/Autodesk/maya-usd/files/3912292/CMakeOutput.log)
[CMakeError.log](https://github.com/Autodesk/maya-usd/files/3912293/CMakeError.log)

**Comment by HdC-adsk** [comment]
There is the following error in CmakeError.log:

`D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\CMakeFiles\CMakeTmp\CheckIncludeFile.c(1): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory [D:\git_pipeline\rez-packages\maya_usd\build\platform-windows\arch==AMD64\maya-usd\CMakeFiles\CMakeTmp\cmTC_2a66c.vcxproj]
`
It looks like a duplicate of https://github.com/Autodesk/maya-usd/issues/30, which has had some instructions from Hamed on how to fix the error. Can you try it?

**Comment by fowlertADSK** [comment]
Can you clarify what variant you are trying to build?  The build log here says RelWithDebInfo but the cmake Output/Error logs look like it’s building Debug.

**Comment by HdC-adsk** [comment]
Sorry I linked with the wrong issue. What I meant to link is https://github.com/Autodesk/maya-usd/issues/86. Another thing to check is whether you missed the arguments when building USD?

`--build-args boost,"--with-date_time --with-thread --with-system --with-filesystem"`

**Comment by BigRoy** [comment]
> Can you clarify what variant you are trying to build? The build log here says RelWithDebInfo but the cmake Output/Error logs look like it’s building Debug.

My arguments are basically like this:
```
    cmd = ["python", "build.py",
           "--generator", cmake_generator, 
           #"-T", cmake_target_platform, 
           "--maya-location", maya_location,
           "--devkit-location", os.path.join(maya_devkit, "devkitBase"),
           "--pxrusd-location", usd_root,
           "--build-location", build_root,
           "--install-location", install_path,
           os.path.join(build_root, "tmp")]
```
So I assume it' sbuilding whatever is the default, likely `RelWithDebInfo`.

> Sorry I linked with the wrong issue. What I meant to link is #86. Another thing to check is whether you missed the arguments when building USD?

Ah, I think that might be a part of this. I did pass along those flags to the build of USD but I recently changed how I did so... maybe there's something wrong with the formatting. Will investigate this tomorrow. Thanks!

**Comment by BigRoy** [comment]
Rebuilt USD but still same error, the USD build:
```
Building with settings:
  USD source directory          D:\git_pipeline\rez-packages\usd\19.11\build\platform-windows\PySide\USD-19.11
  USD install directory         D:\rez_packages\usd\19.11\platform-windows\PySide
  3rd-party source directory    D:\rez_packages\usd\19.11\platform-windows\PySide\src
  3rd-party install directory   D:\rez_packages\usd\19.11\platform-windows\PySide
  Build directory               D:\rez_packages\usd\19.11\platform-windows\PySide\build
  CMake generator               Visual Studio 15 2017 Win64
  Downloader                    curl

  Building                      Shared libraries
    Config                      Release
    Imaging                     On
      Ptex support:             On
      OpenImageIO support:      On
      OpenColorIO support:      On
      PRMan support:            Off
    UsdImaging                  On
      usdview:                  On
    Python support              On
    Documentation               Off
    Tests                       Off
    Alembic Plugin              On
      HDF5 support:             On
    Draco Plugin                Off
    MaterialX Plugin            On
    Maya Plugin                 Off
    Katana Plugin               Off
    Houdini Plugin              Off

  Dependencies                  boost, TBB, HDF5, OpenEXR, Alembic, MaterialX, Ptex, GLEW, OpenSubdiv, JPEG, TIFF, PNG, OpenImageIO, OpenColorIO
  Build arguments               boost: --with-date_time --with-thread --with-system --with-filesystem
STATUS: Installing boost...
STATUS: Installing TBB...
STATUS: Installing HDF5...
STATUS: Installing OpenEXR...
STATUS: Installing Alembic...
STATUS: Installing MaterialX...
STATUS: Installing Ptex...
STATUS: Installing GLEW...
STATUS: Installing OpenSubdiv...
STATUS: Installing JPEG...
STATUS: Installing TIFF...
STATUS: Installing PNG...
STATUS: Installing OpenImageIO...
STATUS: Installing OpenColorIO...
STATUS: Installing USD...

Success! To use USD, please ensure that you have:

    The following in your PYTHONPATH environment variable:
    D:\rez_packages\usd\19.11\platform-windows\PySide\lib\python

    The following in your PATH environment variable:
    D:\rez_packages\usd\19.11\platform-windows\PySide\bin
    D:\rez_packages\usd\19.11\platform-windows\PySide\lib


All 1 build(s) were successful.
```
Note the build arguments for boost. Looks alright?

Same error on building `maya-usd` again. 
The variant being build is indeed `RelWithDebInfo`. It still fails with also explicitly the Visual Studio generator passed along. 

For sake of testing I've switched back to the latest `dev` branch which now has the PR merged into it. That however, seems to run into the same issue now. What exactly has changed compared with before I'm not sure. It must be something with the build process.

[build_output.txt](https://github.com/Autodesk/maya-usd/files/3913205/build_output.txt
[build_log.txt](https://github.com/Autodesk/maya-usd/files/3913200/build_log.txt)
[CMakeError.log](https://github.com/Autodesk/maya-usd/files/3913202/CMakeError.log)
[CMakeOutput.log](https://github.com/Autodesk/maya-usd/files/3913203/CMakeOutput.log)

I'll probably need a good sleep and look at it again. Maybe it's a fishy typo. Are those boost build arguments for USD being parsed correctly? How can I confirm?

**Comment by BigRoy** [comment]
For sake of testing I tried to build against an old commit `6bbd9544593cd45df6ea28a3f2c2511f09686f1d` in `dev` and that one did pass somehow. So it seems it's definitely also related to changes in `maya-usd`?

**Comment by kxl-adsk** [comment]
For USD library, what's the commit you are using? (see https://github.com/Autodesk/maya-usd/blob/dev/doc/DEVELOPER.md)

**Comment by BigRoy** [comment]
@kxl-adsk I recompiled USD with exactly the mentioned commit to be sure, same error when compiling maya-usd.

[build_log.txt](https://github.com/Autodesk/maya-usd/files/3914965/build_log.txt)
[CMakeError.log](https://github.com/Autodesk/maya-usd/files/3914963/CMakeError.log)
[CMakeOutput.log](https://github.com/Autodesk/maya-usd/files/3914964/CMakeOutput.log)

**Comment by BigRoy** [comment]
Now tried to build both USD and Maya-USD without `Rez` and that worked. It must be something with the paths/environment that ends up failing as the compilation/installation of both is very similar. I'll have to investigate what's up with that.

Anyway, I checked the playback issue in latest `dev` and it seems the regression is indeed fixed with #139.

**Comment by BigRoy** [comment]
To complete the investigation of the failing build... I patched it with [this fix mentioned here](https://github.com/Autodesk/maya-usd/issues/86#issuecomment-561011247) and then it succeeded the build in `Rez` too.

It seems that in the Rez environment something results in a space in PYTHONPATH or maybe the CMAKE_BINARY_DIR or alike and the command making them think they are separate arguments. Should this be patched?
